src/wms/src/main/java/org/geoserver/wms/legendgraphic/LegendUtils.java

```diff
@@ -636,13 +650,15 @@ public static BufferedImage renderLabel(
         BufferedImage renderedLabel;
         Color labelColor = getLabelFontColor(req);
         if (LegendUtils.isWrap(req)) {
-            // if label is longer than width, word wrap it.
-            Rectangle2D labelBounds = g.getFontMetrics().getStringBounds(label, g);
-            if (labelBounds.getWidth() > req.getWidth()) {
-                FontMetrics fm = g.getFontMetrics();
-                int widthChars = req.getWidth() / fm.stringWidth("m");
-                WordUtils.wrap(label, widthChars, "\n", true);
+            int width;
+            if (getWrapLimit(req) != -1) {
+                width = getWrapLimit(req);
```

Comment:
I only have a minor comment here:
I would probably assign the getWrapLimit(req) to a local variable to avoid calling the method twice.

Reply:
changes applied!